internal/wal/wal_aof.go

```diff
@@ -124,24 +124,42 @@ func (wal *AOF) Init(t time.Time) error {
 	return nil
 }
 
-// WriteEntry writes an entry to the WAL.
-func (wal *AOF) LogCommand(data []byte) error {
-	return wal.writeEntry(data)
+// Log writes a command to the WAL
+func (wal *AOF) Log(data []byte) error {
+	wal.mu.Lock()
+	wal.lastSequenceNo++
+	lsn := wal.lastSequenceNo
+	wal.mu.Unlock()
+
+	// Create command payload with LSN and wire command bytes
+	payload := &CommandPayload{
+		Lsn:         lsn,
+		WireCommand: data, // data is already the wire command bytes
+	}
+
+	// Marshal the payload to bytes using protobuf
+	payloadBytes, err := proto.Marshal(payload)
+	if err != nil {
+		return fmt.Errorf("failed to marshal command payload: %w", err)
+	}
+
+	return wal.writeEntry(payloadBytes)
 }
```

Comment:
What happens if lsn is incremented and assigned but the writeEntry returns an error. Won't WAL have entries with skipped LSN in this case. Is this fine as long as LSN is incrementing or should we check for error here?